File: src/main/java/legacy/kms/chapter01/Audience.java
package legacy.kms.chapter01;

public class Audience {

    private final Bag bag;

    public Audience(Bag bag) {
        this.bag = bag;
    }

    public long buy(Ticket ticket) {
        if(bag.hasInvitation()){
            bag.setTicket(ticket);
            return 0L;
        } else{
            bag.minusAmount(ticket.getFee());
            return ticket.getFee();
        }
    }
}

File: src/main/kotlin/kms/chapter01/Audience.kt
package kms.chapter01

internal class Audience(private val bag: Bag) {

    fun buy(ticket: Ticket): Long {
        return if (bag.hasInvitation()) {
            bag.setTicket(ticket)
            0L
        } else {
            bag.minusAmount(ticket.fee)
            ticket.fee
        }
    }
}
